docs/GlobalAPI.md

```diff
@@ -1089,3 +1089,19 @@ test.each<{a: number; b: number; expected: string; extra?: boolean}>`
   // all arguments are typed as expected, e.g. `extra: boolean | undefined`
 });
 ```
+
+Keep in mind the variables inside the template literal are not type checked, so you have to ensure that their types are correct.
+
+```ts
+import {test} from '@jest/globals';
+
+test.each<{a: number; expected: string}>`
+  a                            | expected
+  ${1}                         | ${'one'}
+  ${2}                         | ${'two'}
+  ${'will not raise TS error'} | ${'three'}
+`('template literal with wrongly typed input', ({a, expected}) => {
+  // all arguments are typed as stated in the generic: `a: number`, `expected: string`
+  // WARNING: `a` is of type `number` but will be a string the the 3rd test case.
+});
+```
```

Comment:
This behaviour is not related to this PR, but since I'm editing the docs, I tried to clarify it 🙂